File: anothercandies/anothercandies.py
import sys


def main():
    data = [int(x.strip()) for x in sys.stdin.readlines() if x.strip()]
    nn, data = data[0], data[1:]
    for ii in range(nn):
        n_children, data = data[0], data[1:]
        candies, data = data[:n_children], data[n_children:]
        print('YES' if sum(candies) % n_children == 0 else 'NO')


if __name__ == '__main__':
    main()

File: anothercandies/anothercandies.cpp
#include <iostream>

using namespace std;

int main()
{
    int n_cases;
    cin >> n_cases;
    for (int ii = 0; ii < n_cases; ++ii)
    {
        unsigned long long n_children;
        cin >> n_children;
        unsigned long long sum = 0;
        for (unsigned long long jj = 0; jj < n_children; ++jj)
        {
            unsigned long long candies;
            cin >> candies;
            sum = (sum + candies) % n_children;
        }
        if (sum)
        {
            cout << "NO" << endl;
        }
        else
        {
            cout << "YES" << endl;
        }
    }
    return 0;
}